src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CodeFixVerifierHelper.cs

```diff
@@ -112,51 +106,23 @@ public static (SourceText? analyzerConfig, IEnumerable<KeyValuePair<OptionKey, o
             analyzerConfig.AppendLine($"[*.{defaultFileExtension}]");
             foreach (var (key, value) in options)
             {
-                var editorConfigStorageLocation = key.Option.StorageLocations.OfType<IEditorConfigStorageLocation2>().FirstOrDefault();
-                if (editorConfigStorageLocation is null)
+                if (value is NamingStylePreferences namingStylePreferences)
                 {
-                    remainingOptions.Add(KeyValuePairUtil.Create<OptionKey, object?>(key, value));
+                    EditorConfigFileGenerator.AppendNamingStylePreferencesToEditorConfig(namingStylePreferences, key.Language!, analyzerConfig);
                     continue;
                 }
 
-                analyzerConfig.AppendLine(editorConfigStorageLocation.GetEditorConfigString(value, optionSet));
-            }
-
-            return (SourceText.From(analyzerConfig.ToString(), Encoding.UTF8), remainingOptions);
-        }
-
-        private sealed class OptionSetWrapper : OptionSet
-        {
-            private readonly Dictionary<OptionKey, object?> _options;
-
-            public OptionSetWrapper(Dictionary<OptionKey, object?> options)
-            {
-                _options = options;
-            }
-
-#if CODE_STYLE
-            public override bool TryGetValue(string key, out string value)
-            {
-                throw new NotImplementedException();
-            }
-#else
-            public override object? GetOption(OptionKey optionKey)
-            {
-                if (!_options.TryGetValue(optionKey, out var value))
+                var editorConfigStorageLocation = key.Option.StorageLocations.OfType<IEditorConfigStorageLocation2>().FirstOrDefault();
+                if (editorConfigStorageLocation is null)
                 {
-                    value = optionKey.Option.DefaultValue;
+                    remainingOptions.Add(KeyValuePairUtil.Create<OptionKey2, object?>(key, value));
+                    continue;
                 }
 
-                return value;
+                analyzerConfig.AppendLine(editorConfigStorageLocation.GetEditorConfigString(value, default!));
```

Comment:
❕ The `default!` here will need to change to either a sentinel non-null instance, or passing `null` and updating the interface method to allow it without a suppression.

Reply:
Sure.

Reply:
This was leading to quite a few cascaded changes, so I am going to do this as a follow-up PR. Added a tracking item to the end of https://github.com/dotnet/roslyn/issues/41462